lib/chef/data_collector.rb

```diff
@@ -115,9 +117,6 @@ def action_collection_registration(action_collection)
       # (see EventDispatch::Base#run_started)
       #
       def run_started(run_status)
-        Chef::DataCollector::ConfigValidation.validate_server_url!
-        Chef::DataCollector::ConfigValidation.validate_output_locations!
-
```

Comment:
Moved these validations from here and added into `run_start` so that it can pre-validate the when run going to start. As the event sequence defined `run_start` called at step 4 and `run_started` called at step 9 https://github.com/chef/chef/blob/master/docs/dev/design_documents/data_collector.md#relevant-event-sequence

Reply:
Reverted this as integrations specs are failing due to in `run_start` we have not finalized which mode would be enabled so every time are validating `validate_server_url!`